### PR TITLE
fix: escape single quotes in enum values to prevent SQL syntax errors

### LINKED ISSUE #41371: Unescaped quoted enum
# Bug report

## Describe the bug

Enum definitions with quotes raise errors on enum creation or update, requiring manual escape with double quotes right inside the input.

## Screenshots

Image

Same with description and values

Image

### PR DESCRIPTION
**Pull request by lviffy**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Bug fix

## What is the current behavior?

Fixes #41371

Enum values containing single quotes (e.g., `new_t'ype` or `don't`) cause SQL syntax errors when creating or updating enumerated types in Studio.

## What is the new behavior?

Single quotes in enum values are now properly escaped (`'` → `''`) before being sent to PostgreSQL.

## Additional context

- Added [escapeSqlString](cci:1://file:///home/luffy/Projects/supabase/apps/studio/data/enumerated-types/enumerated-type-update-mutation.ts:25:2-26:80) helper to mutation files
- Added 7 unit tests
- Build passes ✅

## Summary by CodeRabbit

* **Bug Fixes**
  * Ensure enumerated type operations safely escape SQL string literals so values and descriptions with quotes are handled correctly.

* **Tests**
  * Added thorough tests covering creation, updates, renames, and edge cases for enumerated types.
  * Improved test environment by providing persistent storage support for browser-like tests.

* **Chores**
  * Adjusted test setup execution order for more reliable test initialization.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/studio/vitest.config.ts [hunk 1]
```diff
@@ -25,9 +25,9 @@ export default defineConfig({
     globals: true,
     environment: 'jsdom', // TODO(kamil): This should be set per test via header in .tsx files only
     setupFiles: [
-      resolve(dirname, './tests/vitestSetup.ts'),
       resolve(dirname, './tests/setup/polyfills.ts'),
       resolve(dirname, './tests/setup/radix.js'),
+      resolve(dirname, './tests/vitestSetup.ts'),
```
[the remark this answers is not in the document]
Comment (lviffy) [reply apps/studio/vitest.config.ts:30]: The reordering is necessary because vitestSetup.ts imports MSW which calls localStorage.getItem() during module initialization. The localStorage mock in polyfills.ts must be defined before MSW is imported, otherwise tests fail with "localStorage.getItem is not a function". This is a required infrastructure change to support the new unit tests.
